dimensiondata/src/test/java/org/jclouds/dimensiondata/cloudcontrol/compute/function/DatacenterToLocationTest.java

```diff
@@ -0,0 +1,95 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jclouds.dimensiondata.cloudcontrol.compute.function;
+
+import com.google.common.collect.ImmutableSet;
+import org.easymock.EasyMock;
+import org.easymock.EasyMockSupport;
+import org.jclouds.dimensiondata.cloudcontrol.domain.Backup;
+import org.jclouds.dimensiondata.cloudcontrol.domain.ConsoleAccess;
+import org.jclouds.dimensiondata.cloudcontrol.domain.Datacenter;
+import org.jclouds.dimensiondata.cloudcontrol.domain.Hypervisor;
+import org.jclouds.dimensiondata.cloudcontrol.domain.Monitoring;
+import org.jclouds.dimensiondata.cloudcontrol.domain.Networking;
+import org.jclouds.dimensiondata.cloudcontrol.domain.Property;
+import org.jclouds.domain.Location;
+import org.jclouds.domain.LocationBuilder;
+import org.jclouds.domain.LocationScope;
+import org.jclouds.location.suppliers.all.JustProvider;
+import org.testng.annotations.BeforeTest;
+import org.testng.annotations.Test;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Set;
+
+import static org.easymock.EasyMock.createNiceMock;
+import static org.easymock.EasyMock.expect;
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertNotNull;
+
+@Test(groups = "unit", testName = "DatacenterToLocationTest")
+public class DatacenterToLocationTest {
```

Comment:
In jclouds we test those kind of functions already using a slightly different pattern - see https://github.com/jclouds/jclouds/blob/master/providers/digitalocean2/src/test/java/org/jclouds/digitalocean2/compute/functions/RegionToLocationTest.java for a great example